hypervisor/src/cpu.rs

```diff
@@ -285,10 +285,16 @@ pub trait Vcpu: Send + Sync {
     ///
     #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
     fn get_one_reg(&self, reg_id: u64) -> Result<u64>;
+    #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
     ///
     /// Retrieve the current cpu states. This function is necessary to snapshot the VM
     ///
     fn cpu_state(&self) -> Result<CpuState>;
+    #[cfg(target_arch = "x86_64")]
+    ///
+    /// Retrieve the current cpu states. This function is necessary to snapshot the VM
+    ///
+    fn cpu_state(&self, msr_list: &MsrList) -> Result<CpuState>;
```

Comment:
Would it be possible to keep the prototype the same between the architectures and have the implementation get the MSRs rather than pass it in. I think that would be nicer.

Reply:
I think that's what we agreed above with @sameo. I'm working on it.